components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -718,7 +718,10 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             throw new ResponseError(ErrorCodes.PERMISSION_DENIED, "Cannot (re-)start a deleted workspace.");
         }
         const userEnvVars = this.userDB.getEnvVars(user.id);
-        let projectEnvVarsPromise = this.internalGetProjectEnvVars(workspace.projectId);
+        const projectEnvVarsPromise = this.internalGetProjectEnvVars(workspace.projectId);
+        const projectPromise = workspace.projectId
+            ? this.projectDB.findProjectById(workspace.projectId)
+            : Promise.resolve(undefined);
```

Comment:
await syntax?

Reply:
Yes, I'm creating a promise here and await it below, especially after the `mayStartPromise` which takes the longest time.